### DISCUSSION TITLE
A home purchase contract

### DISCUSSION THREAD

**Discussion by icolomina** [stellar/sorobounty-spectacular#7] [General]
Hi everyone,
I would like to share my little contribution to sorobounty. 

- **https://github.com/icolomina/home_purchase**:  It is a contract by which a buyer and a seller could interact to make a house purchase transaction.
- **https://dev.to/icolomina/creating-a-dapp-using-php-and-a-house-purchase-soroban-smart-contract-38f1**: This is a post tutorial about how we could build a custodial dapp using php-stellar-sdk soroban support to deploy, install and interact with the contract.

**Comment by briwylde08** [comment]
Hi there! Thank you so much for your submission! We’ll review it and be in contact with you within the next week. 🎉

**Comment by Julian-dev28** [comment]
@icolomina 

Hey thanks for the submission! Great implementation of using PHP with our contracts

A couple of things here:
1. Are you able to add how to deploy the contracts using the cli instead of referencing the deployment doc?
2. Can you do a spell check on both the repo and article? I noticed a few errors throughout reading them

Thanks!

**Reply by Julian-dev28** [reply]
@icolomina I'm really liking the tutorial however this its been awhile since I have used PHP and I assume that not every reader would know which commands to run to invoke the contract using it. 

Are you able to take the example PHP and reference it as a file and then illustrate how to run this file? 

ex:

"If you you want to invoke the contract, follow the steps below: 

First, create a file named`contract.php` with the following:

```
code here
```

Then run the file using:

```
command goes here
```

**Reply by icolomina** [reply]
Hey, sure.
I have a question about point 1. You want me to show readers how to deploy the contract using soroban cli deploy, don't you?

**Reply by icolomina** [reply]
By the way, i come back home from holidays next monday and i will do the tutorial changes. Do you need it before ?

**Reply by briwylde08** [reply]
You're absolutely fine to make the changes when you're back!

**Comment by Julian-dev28** [comment]
> Hey, sure. I have a question about point 1. You want me to show readers how to deploy the contract using soroban cli deploy, don't you?

Exactly.  A simple example with the cli command would be great

**Comment by briwylde08** [comment]
Hi @icolomina! Let me know when you've had a chance to make the submission changes!

**Reply by icolomina** [reply]
Hey,
I've just added in the tutorial how to create users using soroban config and how to deploy a contract using "soroban contract deploy". I am going to review the grammar. 
About the php code, i would need a couple of days more if you do not mind.

Thanks

**Reply by icolomina** [reply]
Sorry, I was confused. The changes about the cli deploy example refer to the contract readme not to the tutorial. I've just added it.

**Comment by icolomina** [comment]
Hey, I've just made the changes:

1.- I've reviewed the grammar of the tutorial and the readme 
2.- I've added an example about how to deploy the contract using soroban-cli on the contract readme
3.- I've added a section on the tutorial (before Last words) which shows how to init a composer project, install the php stellar sdk, create a php file into the src folder and execute it.

I hope it helps to improve the contribution

**Comment by briwylde08** [comment]
Hi there @icolomina! Thank you for your submission! [🎉](https://emojipedia.org/party-popper/)

We've reviewed it (thanks for the discussion and edits!) and given our guidelines, your submission may qualify for an award of $650 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

Thanks again! Please let us know if you have any questions.

Best,
Bri

**Reply by icolomina** [reply]
Hey, Glad to hear it. I am going to fill it asap.

Just a question, I remember I filled that form for sorobanathon. I have to fill out the form again even if I did it before, haven't I ?

**Reply by briwylde08** [reply]
Hi there! Sorry to get back to you so late- was a holiday in the US so we had a long weekend! But yeah, if you could fill it out again, that would be great. It's pretty much the same form but it helps us keep track of all the submission. Thank you!!

**Reply by icolomina** [reply]
Ok, It's done. 
I've attached teh same W-BEN form as I did for sorobanathon

Many Thanks

**Comment by briwylde08** [comment]
Thank you for providing the requested information! You’ve passed KYC and we’ve logged your award payment of $650 worth of XLM, which will be sent to the wallet you designated in the next week.

Thanks again and please let me know if you have any questions!

Best,
Bri
